### PR TITLE
Client buffer mapping

### PR DESCRIPTION
**Pull request by jdashg**

### CODE REVIEW

File: design/buffer-mapping.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Buffer Mapping
+
+For each Buffer with MAP_READ or MAP_WRITE usage, an IPC implemention maintains a parallel
+shared-memory buffer for use with immediate (non-round-trip) mapping.
+
+## WebGLBuffer.map
+
+`ArrayBuffer? WebGLBuffer.map(WebGPUMapFlags flags);`
```
Comment (dmikis) [inline design/buffer-mapping.md]: `WebGPU`?

File: design/buffer-mapping.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+For a Buffer with the MAP_READ usage, the implementation will, upon completion of the
+final pending fence, map the underlying buffer, copy from the underlying buffer into the
+shared-memory buffer, and unmap. This means the shared-memory buffer is ready to be
+mapped.
+
+For a Buffer with MAP_WRITE usage, mapping for WRITE requires that all submitted reads
+from the Buffer have also been confirmed completed. (via Fences)
+
+## WebGLBuffer.unmap
```
Comment (dmikis) [inline design/buffer-mapping.md]: Same here.

Comment (jdashg) [reply design/buffer-mapping.md]: Haha, good catch!

File: design/sketch.webidl [hunk 3]
```diff
@@ -453,13 +455,14 @@ interface WebGPUCommandEncoder {
 
 // Fence
 interface WebGPUFence {
-    u32 getValue();
+    bool wait(u64 timeout);
+    Promise then();
```
Comment (kainino0x) [inline design/sketch.webidl]: nits: `bool`->`boolean`; `Promise then()`->something like [this](https://github.com/gpuweb/gpuweb/pull/49/files#diff-7d154c8abbb1fb80041e213976224980R15)

**Comment by jdashg** [comment]
Since there's been no comments here in a week, let's merge this.

**Pull request merged**